### PR TITLE
Removing UIMnagaer dependencies from the debugger infrastructure

### LINKED ISSUE #15449: DebugSession MUST NOT refer to UIManager
UIManager will be gone soon.

```
resumeProcess
	"Make sure the interrupted process is restored properly and restart the low space handler"

	interruptedProcess isTerminated
		ifFalse: [
			errorWasInUIProcess
				ifTrue: [ UIManager default resumeUIProcess: interruptedProcess ]
				ifFalse: [ interruptedProcess resume ]].

	"restart low space handler"
	Smalltalk installLowSpaceWatcher
```

It means that we should find a better way to resume the UI process.

### PR DESCRIPTION
**Pull request by StevenCostiou**
- cleans the DebugSession
- cleans Oups (almost, I did not touch two methods about warnings yet)

Debuggers are now responsible to handle specific backend requirements:
- graphical debuggers cannot be used if the system is not in graphical mode
- the spec debugger now asks its application backend to know if there are specific actions to take before opening

The PR alone kills the spec debugger.
The debugger can only work after a dedicated PR taking into account the modifications of this PR. 

This PR should:
- make a lot of tests fail (for now)
- replace https://github.com/pharo-project/pharo/pull/15759
- fix https://github.com/pharo-project/pharo/issues/15449 
- not be merged
- be reviewed

### CODE REVIEW

File: src/Debugger-Oups/OupsDebuggerSystem.class.st [hunk 1]
```diff
@@ -83,21 +82,13 @@ OupsDebuggerSystem >> logDebugRequest: aDebugRequest withUIManager: aCommandLine
...
 	aDebugRequest debugSession logStackToFileIfNeeded.
 	self performPreDebugActionsIn: aDebugRequest.
-	self spawnNewUIProcessIfNecessary: aDebugRequest.
 	self ensureExceptionIn: aDebugRequest debugSession.
-
-	"Schedule the debugger opening in a deferred UI message to address redraw problems after opening a debugger e.g. from the testrunner."
-	debuggerOpeningStrategy := self debuggerSelectionStrategy.
-	self defaultUIManager defer: [
-		debuggerOpeningStrategy openDebuggerForSession:
-			aDebugRequest debugSession ].
```
Comment (guillep) [inline src/Debugger-Oups/OupsDebuggerSystem.class.st:100]: Could you comment somewhere in the PR where is the code replacing this functionality? Is it in Spec? Somewhere else? is it WIP?

**Review by guillep — APPROVED** [review]
Certainly much cleaner!

I'll put the PR as draft to avoid accidental merges :)

File: src/Debugger-Oups/OupsDebuggerSystem.class.st (hunk 1 above)
Comment (StevenCostiou) [reply src/Debugger-Oups/OupsDebuggerSystem.class.st:100]: Sorry, it is there https://github.com/pharo-spec/NewTools/pull/667
The spec debugger asks its application for the used backend, which in turn gives back a debug session specialized for that backend (Morphic or Gtk).
The specialized debug session is now handling the removed code that you pointed out.

It also means that if I want to create a new debugger, and if it should work with Morphic, I also have to ask the backend and the specialized session what to do.

Comment (StevenCostiou) [reply src/Debugger-Oups/OupsDebuggerSystem.class.st:100]: In other words: Morphic affairs are now Morphic affairs only.

**Comment by Ducasse** [comment]
This is really good to see some improvements there!
Thanks a lot.

**Comment by Ducasse** [comment]
@guillep @StevenCostiou is this ready to go?

**Comment by StevenCostiou** [comment]
This is too complicated and now we're frozen...
It will be for the early Pharo13

**Comment by Ducasse** [comment]
I moved it to P13 and relaucnhed the tests

**Comment by StevenCostiou** [comment]
Thanks!
It is linked to https://github.com/pharo-spec/NewTools/pull/667 which is now in conflict.
The good part is that the newtools changes may not be reqeuired anymore, I need to check.

**Comment by Ducasse** [comment]
All the tests are passing on windows so I will merge it.

**Comment by StevenCostiou** [comment]
Thanks!
Last time I checked tests were not good :)
This is cool that everything is fixed.

**Comment by astares** [comment]
Thanks for the rollback @jecisc 

Looks like more work is required from  @StevenCostiou 

This change froze the image when you already have put a simple halt into your code or call a non-existing message on a class (leading to a DNU).

**Comment by jecisc** [comment]
We were missing some code from NewTools. This should work now and the change is again present in the image

**Comment by StevenCostiou** [comment]
> Thanks for the rollback @jecisc
> 
> Looks like more work is required from @StevenCostiou
> 
> This change froze the image when you already have put a simple halt into your code or call a non-existing message on a class (leading to a DNU).

Not really, it was written on the PR description: do not merge!
It was waiting for a new tools change, that we integrated earlier this afternoon.

**Pull request merged**